File: packages/NaverLoginSDK/android/src/main/kotlin/net/lagerstroemia/naver_login_sdk/api/NaverLoginSdkConstant.kt
package net.lagerstroemia.naver_login_sdk

object NaverLoginSdkConstant {
    const val channelNameMethod: String = "naver_login_sdk_method"
    const val channelNameEvent: String = "naver_login_sdk_event"
    const val loginBroadcastReceiver: String = "com.lagerstroemia.LOGIN_RESULT"

    object Key {
        const val initialize: String = "initialize"
        const val authenticate: String = "authenticate"
        const val logout: String = "logout"
        const val release: String = "release"
        const val profile: String = "profile"
        const val refresh: String = "refresh"

        const val version: String = "version"
        const val tokenType: String = "tokenType"
        const val accessToken: String = "accessToken"
        const val refreshToken: String = "refreshToken"
        const val expireAt = "expireAt"     // accessTokenExpireDate

        object NaverLoginEventCallback {
            const val onSuccess: String = "onSuccess"
            const val onFailure: String = "onFailure"
            @Deprecated(message = "not use anymore, only use Failure") const val onError: String = "onError"
        }
    }

    object Value {
        const val clientId: String = "clientId"
        const val clientSecret: String = "clientSecret"
        const val clientName: String = "clientName"

        object OAuthLoginCallback {
            const val errorCode: String = "errorCode"
            const val message: String = "message"
        }
    }
}

File: packages/NaverLoginSDK/ios/naver_login_sdk/Sources/naver_login_sdk/Api/NaverLoginSdkConstant.swift
//
//  NaverLoginSdkConstant.swift
//  naver_login_sdk
//
//  Created by Lagerstroemia on 11/17/24.
//

import Foundation

struct NaverLoginSdkConstant {
    static let channelNameMethod: String = "naver_login_sdk_method"
    static let channelNameEvent: String = "naver_login_sdk_event"
    
    struct Key {
        static let initialize: String  = "initialize"
        static let authenticate: String = "authenticate"
        static let logout: String = "logout"
        static let release: String = "release"
        static let profile: String = "profile"
        static let refresh: String = "refresh"
        
        static let version: String = "version"
        static let tokenType: String = "tokenType"
        static let acessToken: String = "accessToken"
        static let refreshToken: String = "refreshToken"
        static let expireAt: String = "expireAt"
        
        struct NaverLoginEventCallback {
            static let onSuccess: String = "onSuccess"
            static let onFailure: String = "onFailure"
            static let onError: String = "onError"
        }
    }
    
    struct Value {
        static let urlScheme: String = "urlScheme"
        static let clientId: String = "clientId"
        static let clientSecret: String = "clientSecret"
        static let clientName: String = "clientName"
    }
}
